### PR TITLE
Do not return value if the database id used for `last-used-native-database-id` does not exist (anymore)

### PR DESCRIPTION
**Pull request by nemanjaglumac**
This PR is a follow up after #38986.

### Description

It is quite possible that a user persists a database (id), and that this database gets deleted afterwards.
Even though the frontend handles this pretty well (which is confirmed by the existing E2E test), we wanted to follow up with a custom `:getter` for this particular setting that would return `nil` if the persisted database doesn't exist anymore.

This PR tries to achieve exactly that.
It updates and expands the previously written backend unit tests to reflect this new condition.

### How to verify
At the very minimum, all tests should pass.

You can do this manually:
1.  Send a `PUT /api/setting/last-used-native-database-id`. For the `value` use a database id value that doesn't exist (e.g. 999)
2.  Call `GET /api/session/properties` and make sure that `last-used-native-database-id` is `null`

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by ranquild — APPROVED** [review]
Please wait for the BE approval

File: test/metabase/models/user_test.clj
Comment (metamben) [inline test/metabase/models/user_test.clj:543]: This is a bit brittle. Can we set a negative value? (Normal DB IDs are positive.)

File: test/metabase/models/user_test.clj
Comment (metamben) [inline test/metabase/models/user_test.clj:566]: It looks as if we tried to return to the original settings. If that's indeed the intent, then it's better done in a `finally` form.
```suggestion
          (try
            (user/last-used-native-database-id! id1)
            (mt/with-test-user :crowberto
              (let [old-db-id (user/last-used-native-database-id)]
                (try
                  (user/last-used-native-database-id! id2)
                  (is (= (user/last-used-native-database-id) id2))
                  (mt/with-test-user :rasta
                    (is (= (user/last-used-native-database-id) id1)))
                  (finally
                    (user/last-used-native-database-id! old-db-id))))
             (finally
               (user/last-used-native-database-id! old-db-id)))))))
```

**Review by metamben — APPROVED** [review]
LGTM, just a couple of small things that might make the test more robust.

File: test/metabase/models/user_test.clj
Comment (nemanjaglumac) [reply test/metabase/models/user_test.clj]: I ended up using `mt/discard-setting-value` to simplify this a bit.
https://github.com/metabase/metabase/pull/39269/commits/2d7328066b11570760b8941ea5b0831b3010763d

Please see [this Slack convo](https://metaboat.slack.com/archives/CKZEMT1MJ/p1709558933798589) for context.

**Review by metamben — APPROVED** [review]
LGTM. It seems it can be further simplified by using `with-temporary-setting-values`.

**Pull request merged**
